src/gui/providers/gdal/qgsgdalsourceselect.h

```diff
@@ -50,6 +50,24 @@ class QgsGdalSourceSelect : public QgsAbstractDataSourceWidget, private Ui::QgsG
     void radioSrcProtocol_toggled( bool checked );
     void cmbProtocolTypes_currentIndexChanged( const QString &text );
 
+  protected:
+
+    /**
+     * Sets the dialog map canvas
+     * \see mapCanvas()
+     *
+     * \since QGIS 3.24
+     */
+    void setMapCanvas( QgsMapCanvas *mapCanvas ) override;
```

Comment:
See comment in QgsAbstractDataSourceWidget headers -- we don't need to make setMapCanvas and mapCanvas **pure** virtual (just virtual, not pure), and then we can avoid touching all these other non-WCS subclasses